src/docfx/lib/log/Telemetry.cs

```diff
@@ -73,6 +78,24 @@ public static void SetRepository(string? repo, string? branch)
             s_telemetryClient.Context.GlobalProperties["Branch"] = s_branch;
         }
 
+        public static void TrackDocfxConfig(string fileName, string docsetName, JObject docfxConfig)
+        {
+            var newValue = JsonUtility.DeepClone(docfxConfig) as JObject;
+            var docfxConfigTelemetryValue = JsonUtility.Serialize(newValue!);
+            var hashCode = HashUtility.GetMd5Hash(docfxConfigTelemetryValue);
+            if (docfxConfigTelemetryValue.Length > MaxEventPropertyLength)
+            {
+                TryRemoveNestedObject(newValue!);
```

Comment:
The `!` here will cause an `NRE` if the config is not a `JObject`, in which case we can skip the fallback.